### PR TITLE
feat(holocronModuleWrapper): make moduleState opt out

### PR DESCRIPTION
**Pull request by code-forger**
## Motivation and Context
This can prevent unnecessary 'toJS()' calls and re-renders which is a major performance point for large applications

## How Has This Been Tested?
unit tests

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] My changes are in sync with the code style of this project.
- [x] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for Holocron users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using Holocron?
N/A

### CODE REVIEW

File: packages/holocron/__tests__/holocronModule.spec.jsx [hunk 1]
```diff
@@ -474,6 +474,41 @@ describe('holocronModule', () => {
...
+      reducer,
+    })(({ moduleState }) => <div>{JSON.stringify(moduleState)}|{typeof moduleState}</div>);
+    const mockStore = createStore((state) => state, fromJS({ modules: { 'mock-module': { mockKey: 'mockValue' } } }));
+    const component = renderer.create(
+      <Provider store={mockStore}>
+        <Module />
+      </Provider>
+    );
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
```
Comment (10xLaCroixDrinker) [inline packages/holocron/__tests__/holocronModule.spec.jsx:490]: Can we use inline snapshots? These are really small

File: packages/holocron/src/holocronModule.jsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ export default function holocronModule({
   shouldModuleReload,
   loadModuleData,
   mergeProps,
-  options = {},
+  options = { provideModuleState: true },
```
Comment (10xLaCroixDrinker) [inline packages/holocron/src/holocronModule.jsx:97]: This is breaking to anyone passing any options today. If a user is passing any options at all this default value will not be present.

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

**Comment by 10xLaCroixDrinker** [comment]
This PR also requires a documentation update

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 1 above)
Comment (code-forger) [reply packages/holocron/__tests__/holocronModule.spec.jsx:490]: Adding inline snapshots caused prettier to re-format this file, i'm investigating why it was ignore our eslint config

File: packages/holocron/src/holocronModule.jsx (hunk 2 above)
Comment (code-forger) [reply packages/holocron/src/holocronModule.jsx]: addressed b9ed653

File: packages/holocron/src/holocronModule.jsx [hunk 3]
```diff
@@ -94,7 +94,7 @@ export default function holocronModule({
   shouldModuleReload,
   loadModuleData,
   mergeProps,
-  options = {},
+  options = { },
```
Comment (Matthew-Mallimo) [inline packages/holocron/src/holocronModule.jsx:97]:
```suggestion
  options = {},
```
I know this is dumb but it bothers me 😄

File: packages/holocron/__tests__/holocronModule.spec.jsx (hunk 1 above)
Comment (code-forger) [reply packages/holocron/__tests__/holocronModule.spec.jsx]: addressed 8b66203

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**